### PR TITLE
Advanced search

### PR DESCRIPTION
**Pull request by huyxgit**
Screen Shot 2021-04-23 at 2 12 48 AM

### CODE REVIEW

**Comment by huynguyez** [comment]
@humphd this question is a bit further when this advanced search component completed: should we wire up this search dialog to the searchButton on Nav bar instead of redirecting to search-page right away?

**Comment by humphd** [comment]
@huynguyez good question.  At some point (later) we might want to have a way to search from the main page without moving over to this page.  For now, let's have a way to get to the advanced search from the search page, but not have that be the default.

**Comment by huynguyez** [comment]
@HyperTHD

**Comment by izhuravlev** [comment]
@huynguyez do you need help? I can hop on a call in about 2 hours and work on this so we can merge it.

**Comment by huynguyez** [comment]
@izhuravlev sure, still problems with the dialog, state

File: src/web/src/components/AdvancedSearchDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import FormControl from '@material-ui/core/FormControl';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import React, { useState } from 'react';
+
+interface Props {
+  setOpenDialog: Function;
+  openDialog: boolean;
+  searchTerm: string;
```
Comment (HyperTHD) [inline src/web/src/components/AdvancedSearchDialog.tsx:14]: Same issue here, if this is just the search text, you can just use `text` from `useSearchValue()`

**Review by HyperTHD — CHANGES_REQUESTED** [review]
Just a preliminary review. There's stuff other stuff I need to look at but this is a few stuff I caught

File: src/web/src/components/AdvancedSearchDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        <p>Keyword:</p>
+        <form
+          onSubmit={(e) => {
+            if (text) {
+              onSubmitHandler(e);
+            }
+            handleClose();
+          }}
+        >
+          <input
```
[the remark this answers is not in the document]
Comment (huynguyez) [reply src/web/src/components/AdvancedSearchDialog.tsx:165]: @HyperTHD right, but what if user wanna change keyword to something else?

[the remark this answers is not in the document]
Comment (huynguyez) [reply src/web/src/components/AdvancedSearchDialog.tsx:165]: so I put an input in this dialog as well

[the remark this answers is not in the document]
Comment (HyperTHD) [reply src/web/src/components/AdvancedSearchDialog.tsx:165]: If the user wants to change the keyword, then they can close the advanced search tab and enter something new.

File: src/web/src/components/AdvancedSearchDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+  const handleClose = () => {
+    props.setOpenDialog(false);
+    setSearchInAuthor(false);
+
+    console.log('close: ' + searchInAuthor);
+  };
+
+  const handleCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInAuthor(e.target.checked);
```
Comment (HyperTHD) [inline src/web/src/components/AdvancedSearchDialog.tsx:111]: This value might need to be added in the SearchProvider, the toggle is not being read by the provider which is why it's not working

**Review by HyperTHD — CHANGES_REQUESTED** [review]
You're probably going to have to update the SearchProvider so that you can access a state toggle value similar to SearchHelp. Look at how SearchHelp's toggle value is implemented in SearchProvider.tsx for reference

File: src/web/src/components/AdvancedSearchDialog.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (huynguyez) [reply src/web/src/components/AdvancedSearchDialog.tsx:165]: @HyperTHD  it seems everything work fine if I remove the "search" button on dialog

**Comment by izhuravlev** [comment]
I can't input text into the search bar - sorry for the delay, we can stay on the call after the meeting and work on it till it's done

**Comment by huynguyez** [comment]
@HyperTHD I tried to store the state in global but it still same. any idea?

**Comment by huynguyez** [comment]
Not work when I store state in Context (search provider). But using states in local component is working as expected. Obviously this should be optimized later

File: src/web/src/components/AdvancedSearchDialog.tsx (hunk 3 above)
Comment (huynguyez) [reply src/web/src/components/AdvancedSearchDialog.tsx]: I tried to store in Search Provider but same

**Review by LuigiZaccagnini — CHANGES_REQUESTED** [review]
Small change.

**Comment by AmasiaNalbandian** [comment]
There is work in progress on the branch right here: 
https://github.com/AmasiaNalbandian/telescope/tree/issue-2110-advancedsearch

I have cherry picked the beginning code for the modal, and have made more progress by implementing the backend for the advanced search from the PR that was added here: https://github.com/Seneca-CDOT/telescope/pull/2617

I am going to close this PR as it is abandoned, and I have picked up the issue.

**Pull request closed without merging**